File: cobalt/backend/presentation/contracts/http/routers/games.py
#  Copyright (C) 2026 ArtoriasCode
#  Author: ArtoriasCode
#  Repository: https://github.com/ArtoriasCode/cobalt
#  SPDX-License-Identifier: AGPL-3.0-or-later

from abc import abstractmethod, ABC
from typing import Any

from presentation.contracts.http.routers import AbstractHttpRouter


class AbstractHttpGamesRouter(AbstractHttpRouter, ABC):
    """
    Abstract HTTP router for games-related operations.
    """

    @abstractmethod
    async def get_page(self, *args: Any, **kwargs: Any) -> Any:
        """
        Gets a paginated list of games.

        Parameters:
        - *args: Positional arguments.
        - **kwargs: Keyword arguments.

        Returns:
        - Any: Response object.
        """
        ...

    @abstractmethod
    async def get_one(self, *args: Any, **kwargs: Any) -> Any:
        """
        Gets an existing game by ID.

        Parameters:
        - *args: Positional arguments.
        - **kwargs: Keyword arguments.

        Returns:
        - Any: Response object.
        """
        ...

File: cobalt/frontend/src/contracts/services/api/http/games.ts
/*
 * Copyright (C) 2026 Artorias
 * Author: Artorias
 * Repository: https://github.com/artorias-developer/cobalt
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import type { GameEntity, GamesPageEntity, GamesPageRequest } from "@/types"

/**
 * Interface for games API service operations.
 * Defines contract for managing games.
 * Implementation-agnostic - not tied to any specific HTTP client.
 */
export interface IHttpGamesApiService {
  /**
   * Gets a paginated list of games.
   *
   * Parameters:
   * - params: GamesPageRequest object.
   *
   * Returns:
   * - Promise<GamesPageEntity>: Paginated list of games.
   */
  getPage(params: GamesPageRequest): Promise<GamesPageEntity>

  /**
   * Gets an existing game by ID.
   *
   * Parameters:
   * - gameId: Game ID.
   *
   * Returns:
   * - Promise<GameEntity>: Game entity.
   */
  getOne(gameId: number): Promise<GameEntity>
}